File: cdk_integration_tests/src/python/RedShiftSSL/fail__2__.py
from aws_cdk import core
from aws_cdk import aws_redshift as redshift

class MyRedshiftClusterParameterGroupStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define Redshift Cluster Parameter Group with require_ssl parameter
        redshift.CfnClusterParameterGroup(
            self, 'MyRedshiftClusterParameterGroup',
            description='My Redshift Parameter Group',
            parameter_group_family='redshift-1.0',
            parameters=[
                redshift.CfnClusterParameterGroup.ParameterProperty(
                    parameter_name='require_ssl',
                    parameter_value='false'
                )
                # Add other parameters if needed
            ]
        )

app = core.App()
MyRedshiftClusterParameterGroupStack(app, "MyRedshiftClusterParameterGroupStack")
app.synth()

class MyRedshiftClusterParameterGroupStack2(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define Redshift Cluster Parameter Group with require_ssl parameter
        redshift.CfnClusterParameterGroup(
            self, 'MyRedshiftClusterParameterGroup',
            description='My Redshift Parameter Group',
            parameter_group_family='redshift-1.0',
            parameters=[
                redshift.CfnClusterParameterGroup.ParameterProperty(
                    parameter_name='abc',
                    parameter_value='true'
                )
                # Add other parameters if needed
            ]
        )

app = core.App()
MyRedshiftClusterParameterGroupStack2(app, "MyRedshiftClusterParameterGroupStack2")
app.synth()

File: cdk_integration_tests/src/typescript/RedShiftSSL/fail__2__.ts
import * as cdk from 'aws-cdk-lib';
import * as redshift from 'aws-cdk-lib/aws-redshift';

class MyRedshiftClusterParameterGroupStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        // Define Redshift Cluster Parameter Group with require_ssl parameter
        new redshift.CfnClusterParameterGroup(this, 'MyRedshiftClusterParameterGroup', {
            description: 'My Redshift Parameter Group',
            parameterGroupFamily: 'redshift-1.0',
            parameters: [
                {
                    parameterName: 'require_ssl',
                    parameterValue: 'false',
                },
                // Add other parameters if needed
            ],
        });
    }
}

const app = new cdk.App();
new MyRedshiftClusterParameterGroupStack(app, 'MyRedshiftClusterParameterGroupStack');
app.synth();

class MyRedshiftClusterParameterGroupStack2 extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        // Define Redshift Cluster Parameter Group with abc parameter
        new redshift.CfnClusterParameterGroup(this, 'MyRedshiftClusterParameterGroup2', {
            description: 'My Redshift Parameter Group 2',
            parameterGroupFamily: 'redshift-1.0',
        });
    }
}

new MyRedshiftClusterParameterGroupStack2(app, 'MyRedshiftClusterParameterGroupStack2');
app.synth();
